allauth/mfa/middleware.py

```diff
@@ -0,0 +1,56 @@
+from django.contrib import messages
+from django.http import HttpResponseRedirect
+from django.urls import reverse
+from django.utils.deprecation import MiddlewareMixin
+
+from allauth import app_settings
+from allauth.account.adapter import get_adapter
+from allauth.mfa.utils import is_mfa_enabled
+
+
+class RequireMFAMiddleware(MiddlewareMixin):
+    """Force multi-factor authentication for certain users."""
+
+    allowed_urls = [
+        "account_login",
+        "account_change_password",
+        "account_logout",
+        "account_reset_password",
+        "mfa_activate_totp",
+    ]
+
```

Comment:
I would highly recommend adding at least `account_reauthenticate` to the `allowed_urls` list to avoid infinite redirections. If MFA is enforced for a user, not activated yet and your session expire you will encounter the following loop:

- `account_reauthenticate`
- `mfa_activate_totp`
- `account_reauthenticate`
- ...

Adding `account_reauthenticate` to the list will solve that issue. More globally, it might be interesting to review all the URLs (e.g `account_inactive`...)

Reply:
Thanks! That is a very good suggestion. I will also take another closer look at all other URLs. 
I have some deadlines at work this week, but should be able to work on this again next week.

Reply:
I have updated the allowed urls. I had missed quite a few on second thought. The sign-in and sign-up urls don't need to be added in my opinion since the user won't be authenticated on those views yet and hence they are implicitly allowed.

Reply:
Great! Already using it on my side. I hope it will be merged.